### PR TITLE
made OOP to use new host group feature in service hub

### PR DESCRIPTION
**Pull request by heejaechang**
this allow each service hub client (VS) to have its own service hub host process (roslyn OOP process)

also added cancellation check before actually asking asset to VS.

### CODE REVIEW

**Comment by heejaechang** [comment]
@KevinH-MS @panopticoncentral @AArnott can you take a look?

**Comment by heejaechang** [comment]
test vsi please

**Comment by heejaechang** [comment]
make sure roslyn is safe with older service hub bits in vs drop preview 5 or less.

**Comment by heejaechang** [comment]
test vsi please

**Comment by AArnott** [comment]
@IlyaBiryukov do you want to take a look?

**Comment by IlyaBiryukov** [comment]
Looks good.

**Comment by KevinH-MS** [comment]
LGTM, @heejaechang I tagged you on my LUT PR.  We can check these in at the same time.

**Comment by heejaechang** [comment]
@MattGertz can I get approval for RC? tagging @srivatsn as well.

**Comment by heejaechang** [comment]
retest windows_release_unit64_prtest please

**Comment by heejaechang** [comment]
version conflict issue

**Comment by heejaechang** [comment]
failed again due to version conflict.

**Comment by heejaechang** [comment]
retest windows_release_unit64_prtest please

re-trying. doesn't repro on dev box

**Comment by heejaechang** [comment]
retest windows_release_unit64_prtest please

File: src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs [hunk 1]
```diff
@@ -98,5 +102,19 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
...
+
+        private static async Task<Stream> RequestServiceAsync(HubClient client, string serviceName, string hostGroup, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // we can remove these once whole system moved to new servicehub API
+            try
+            {
+                var descriptor = new ServiceDescriptor(serviceName) { HostGroup = new HostGroup(hostGroup) };
+                return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
+            }
+            catch
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs]: can we only catch the specific exception that gets thrown when the ServiceDescriptor route isn't avialable?

File: src/Workspaces/Remote/Core/Services/AssetStorage.cs [hunk 2]
```diff
@@ -35,9 +35,15 @@ public AssetStorage()
             Task.Run(CleanAssetsAsync, CancellationToken.None);
         }
 
-        public AssetSource GetAssetSource(int sessionId)
+        public AssetSource TryGetAssetSource(int sessionId)
         {
-            return _assetSources[sessionId];
+            AssetSource source;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/Core/Services/AssetStorage.cs]: Why is this needed as part of this PR?

Comment (heejaechang) [inline src/Workspaces/Remote/Core/Services/AssetStorage.cs]: changed description.

File: src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs (hunk 1 above)
Comment (heejaechang) [inline src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs]: I dont know all exception that can be thrown.
